bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/items/RollershutterItemTest.java

```diff
@@ -49,4 +57,43 @@ public void setState_stateDecimal050_returnPercent50() {
         assertEquals(new PercentType(50), sut.getState());
     }
 
+    @Test
+    public void setState_stateOpen_returnPercent100() {
+        RollershutterItem sut = new RollershutterItem("Test");
+        State state = OpenClosedType.OPEN;
+        sut.setState(state);
```

Comment:
I did not (yet) come across such usage of types on RollerShutters but I have seen quite some extraordinary combinations of types on items. That's why I thought I create a test for any of those _I_ could imagine that people use them. But if we agree on disallowing certain combinations that's fine for me as well. I just don't know if people are relying on certain combinations already because they were possible until now.

@sjka: No i have not yet checked all callers, but I can have a look at them. But I think the callers should fix it because if we do not return `null` we do not have a change of telling the caller that a conversion has failed. In such a case the caller would have to do a check "is the returned state the same as the input state" and if I have to do such a check I can also do the "null check". What do you think?